avocado/utils/memory.py

```diff
@@ -25,9 +25,86 @@
 
 from . import process
 from . import genio
+from . import wait
 
 
-# Returns total memory in kb
+class MemoryError(Exception):
+
+    """
+    called when memory operations fails
+    """
+    pass
+
+
+def _check_memory_state(block):
+    """
+    Check the given memory block is online or offline
+
+    :param block: memory block id.
+    :type string: like 198
+    :return: 'True' if online or 'False' if offline
+    :rtype: bool
+    """
+    def _is_online():
+        with open('/sys/devices/system/memory/memory%s/state' % block, 'r') as state_file:
+            if state_file.read() == 'online\n':
+                return True
+            return False
+
+    return wait.wait_for(_is_online, timeout=120, step=1) or False
+
+
+def _check_hotplug():
+    """
+    Check kernel support for memory hotplug
+
+    :return: True if hotplug supported,  else False
+    :rtype: 'bool'
+    """
+    if glob.glob('/sys/devices/system/memory/memory*'):
+        return True
+    return False
+
+
+def _is_hot_pluggable(block):
```

Comment:
This helper function is not being used anywhere now. If it's intended to be used by users, it should not have the leading underscore in the name.